zerver/lib/push_notifications.py

```diff
@@ -1348,19 +1348,25 @@ def handle_push_notification(user_profile_id: int, missed_message: Dict[str, Any
         trigger = NotificationTriggers.DIRECT_MESSAGE  # nocoverage
 
     mentioned_user_group_name = None
+    mentioned_user_group_members_count = None
     # mentioned_user_group_id will be None if the user is personally mentioned
     # regardless whether they are a member of the mentioned user group in the
     # message or not.
     mentioned_user_group_id = missed_message.get("mentioned_user_group_id")
 
     if mentioned_user_group_id is not None:
-        user_group = UserGroup.objects.get(
-            id=mentioned_user_group_id, realm_id=user_profile.realm_id
+        user_group = (
+            UserGroup.objects.filter(id=mentioned_user_group_id, realm_id=user_profile.realm_id)
+            .annotate(direct_members_count=Count("direct_members"))
+            .first()
         )
         mentioned_user_group_name = user_group.name
+        mentioned_user_group_members_count = user_group.direct_members_count
```

Comment:
Is it easy to share the counting code with `get_mentioned_user_group` instead? It kinda seems like similar code, but haven't read closely. I guess this one has the advantage that we only need to look up a single group.